src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/IHttpStartLineHandler.cs

```diff
@@ -0,0 +1,9 @@
+using System;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Internal.Http
+{
+    public interface IHttpStartLineHandler
+    {
+        void OnStartLine(HttpMethod method, HttpVersion version, Span<byte> target, Span<byte> path, Span<byte> query, Span<byte> customMethod);
```

Comment:
What is the advantage of doing this string -> enum conversions for methods? There should be no cost for known methods since we're using the same string refs everywhere. If you keep it a string as before, the API looks much cleaner. Having `method` and `customMethod` in the call (and so far from each other 😝) is kinda crummy.